bin/convencommitcheck.bash

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Define valid types and scopes
+VALID_TYPES=("build" "chore" "ci" "docs" "feat" "fix" "perf" "refactor" "style" "test")
+VALID_SCOPES=("dev" "comp" "lib" "fd2" "examples" "school" "none") # Added 'none' for no scope
+
+# Helper function to check if an element is in an array
+elementInArray() {
+    local element
+    for element in "${@:2}"; do
+        [[ "$element" == "$1" ]] && return 0
+    done
+    return 1
+}
+
+# Function to prompt for a value with a default
+promptForValue() {
+    local prompt=$1
+    local default=$2
+    read -p "$prompt [$default]: " value
+    echo "${value:-$default}"
+}
+
+# Function to convert PR title to conventional commit format
+convertToConventionalCommit() {
+    local type=$1
+    local scope=$2
+    local description=$3
+    local body=$4
+    local breaking_change=$5
+    local commit_message="${type}"
+    if [[ "$scope" != "none" ]]; then
+        commit_message="${commit_message}(${scope})"
+    fi
+    commit_message="${commit_message}: ${description}"
+    if [[ "$breaking_change" == "yes" ]]; then
+        commit_message="${commit_message}\n\nBREAKING CHANGE: ${body}"
```

Comment:
Open to discussion:  I don't know if we want the entire body to become the BREAKING CHANGE description.  It seems likely that some PRs will have both non-breaking changes and BREAKING CHANGES to describe.  So maybe we should have a way to pull those apart?  Not sure what it would be.  Any thoughts?

Reply:
> Open to discussion: I don't know if we want the entire body to become the BREAKING CHANGE description. Some PRs will likely have both non-breaking changes and BREAKING CHANGES to describe. So we should have a way to pull those apart. Not sure what it would be. Any thoughts?

That's an interesting one. My thought is to implement an interactive approach that prompts the author to manually input the breaking change description if a breaking change is indicated is my recommendation. Something along the lines of this (We would also have to modify `convertToConventionalCommit()` to accept new parameters as well):
```bash
BREAKING_CHANGE=$(promptForValue "Is this a breaking change (yes/no)" "no")
BREAKING_CHANGE_DESCRIPTION=""
if [[ "$BREAKING_CHANGE" == "yes" ]]; then
    echo "You indicated this is a breaking change. Please provide a specific description of the breaking change."
    read -p "Enter breaking change description: " BREAKING_CHANGE_DESCRIPTION
fi
```
The way that I see it, if it has any amount of breaking changes, it should be treated if it was holistically breaking changes only.

Reply:
> My thought is to implement an interactive approach that prompts the author to manually input the breaking change description if a breaking change is indicated and putting it in the commit body

This sounds like a good approach.  But only if the PR body doesn't already contain a BREAKING CHANGE footer.  If it does, extract it and use that.

Reply:
> The way that I see it, if it has any amount of breaking changes, it should be treated if it was holistically breaking changes only.

Yes.  If a conventional commit has a body and a BREAKING CHANGE footer, it will bump the major version number.  So if a commit contains both non-breaking changes and breaking changes, the semantic version number will be adjusted as a BREAKING CHANGE.

Reply:
> > The way that I see it, if it has any amount of breaking changes, it should be treated if it was holistically breaking changes only.
> 
> Yes. If a conventional commit has a body and a BREAKING CHANGE footer, it will bump the major version number. So if a commit contains both non-breaking changes and breaking changes, the semantic version number will be adjusted as a BREAKING CHANGE.

Perfect! I assume we would then adhere to something like this
```
<type>(<scope>): <short description>

<BODY>

BREAKING CHANGE: <breaking change description>
```

Reply:
Added BREAKING CHANGE checks in PR Body